### PR TITLE
Add constraints for history plot

### PR DESCRIPTION
**Pull request by hrntsm**
## Contributor License Agreement

This repository (``optuna-dashboard``) and [Goptuna](https://github.com/c-bata/goptuna) share common code.
This pull request may therefore be ported to Goptuna.
Make sure that you understand the consequences concerning licenses and check the box below if you accept the term before creating this pull request.

* [x] I agree this patch may be ported to [Goptuna](https://github.com/c-bata/goptuna) by other Goptuna contributors.

## Reference Issues/PRs

Support constraint in optimization history plot like #525 

## What does this implement/fix? Explain your changes.

Both plots for the study alone and comparison plots were made to take into account the constraints.

Screenshot 2023-07-27 at 21 04 04

Screenshot 2023-07-27 at 21 04 47

### CODE REVIEW

File: optuna_dashboard/ts/components/GraphHistory.tsx [hunk 1]
```diff
@@ -411,6 +423,20 @@ const plotHistory = (
...
   }
+  plotData.push({
+    x: infeasibleTrials.map(getAxisX),
+    y: infeasibleTrials.map(
+      (t: Trial): number => target.getTargetValue(t) as number
+    ),
+    name: "Infeasible Trial",
+    marker: {
+      size: markerSize,
+      color: "#cccccc",
```
Comment (c-bata) [inline optuna_dashboard/ts/components/GraphHistory.tsx:434]: [nit] `#cccccc` might be a bit too light in Dark mode. What do you think about using `#666666` in dark mode instead?

Screenshot 2023-07-28 10 00 04
Screenshot 2023-07-28 9 59 58

```suggestion
      color: mode === "dark" ? "#666666" : "#cccccc",
```

Then the history plot will be rendered like below:

Screenshot 2023-07-28 10 07 13

File: optuna_dashboard/ts/components/GraphHistory.tsx [hunk 2]
```diff
@@ -528,7 +562,22 @@ const plotHistoryMultiStudies = (
...
     }
+    infeasiblePlotData.push({
+      x: infeasibleTrials.map(getAxisX),
+      y: infeasibleTrials.map(
+        (t: Trial): number => target.getTargetValue(t) as number
+      ),
+      name: `Infeasible Trial of ${h.study_name}`,
+      marker: {
+        size: markerSize,
+        color: "#cccccc",
```
Comment (c-bata) [inline optuna_dashboard/ts/components/GraphHistory.tsx:573]: ditto.

**Review by c-bata — COMMENTED** [review]
@hrntsm Thank you for your pull request! Looks almost good to me. I left one suggestion though.

**Comment by hrntsm** [comment]
Updated colors. It's easier to see now.

**Review by c-bata — APPROVED** [review]
LGTM!

**Pull request merged**